crates/miden-lib/asm/kernels/transaction/lib/memory.masm

```diff
@@ -672,13 +685,24 @@ export.get_max_foreign_account_ptr
     push.MAX_FOREIGN_ACCOUNT_PTR
 end
 
-#! Sets the memory pointer of the current account data to the native account (2048).
+#! Sets the memory pointer of the current account data to the native account (8192).
 #!
 #! Inputs:  []
 #! Outputs: []
 export.set_current_account_data_ptr_to_native_account
-    push.NATIVE_ACCOUNT_DATA_PTR
-    push.CURRENT_ACCOUNT_DATA_PTR mem_store
+    # get the pointer to the first stack element where pointer to the native account data should be 
+    # stored
+    push.ACCOUNT_STACK_TOP_PTR.MIN_ACCOUNT_STACK_PTR
```

Comment:
nit: I'd split this into two separate `push` instructions:
```
push.ACCOUNT_STACK_TOP_PTR
push.MIN_ACCOUNT_STACK_PTR
```
It took me a little bit of time to figure out that this was actually pushing 2 values onto the stack.